Silicon/ElkhartlakePkg/Library/HeciInitLib/HeciPeiDxe.c

```diff
@@ -494,3 +495,135 @@ HeciArbSvnGetInfoMsg (
   FreePool (ArbSvnGetInfo);
   return Status;
 }
+
+/**
+  The command is used to get the information about OEM keys, such as
+  key hash, revocation status, validity.
+
+  @param[in, out] OemKeyStatus    Status of OEM keys
+
+  @retval EFI_SUCCESS             Command succeeded
+  @retval EFI_UNSUPPORTED         Current ME mode doesn't support this function
+  @retval EFI_DEVICE_ERROR        HECI Device error, command aborts abnormally
+**/
+EFI_STATUS
+HeciGetOemKeyStatus(

```

Comment:
Just wondering..if we need to put this in platform specific implementation?
OEM Key Revocation will be supported in EHL, TGL, ADL and MTL.

If we put this (GetOemKeyStatus/RevokeOemKey) in Common HECI interface, and a non-compliant platform implementation invokes it, it will just receive a   un-implemented HECI error from CSE. Isn't it ?

Reply:
Currently EHL/TGL/CML uses their platform HECI lib implementations.  Agreed, this would be moved to common once these platforms adapts common Heci Library.

Reply:
Got it.